jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -592,4 +631,25 @@ public Completable compose(Consumer<Completable> consumer)
             return completable;
         }
     }
+
+    /**
+     * Invoke a callback failure, handling any {@link Throwable} thrown
+     * by adding the passed {@code failure} as a suppressed with
+     * {@link ExceptionUtil#addSuppressedIfNotAssociated(Throwable, Throwable)}.
+     * @param callback The callback to fail
+     * @param failure The failure
+     * @throws RuntimeException If thrown, will have the {@code failure} added as a suppressed.
+     */
+    static void failed(Callback callback, Throwable failure)
```

Comment:
I don't particularly like to have another method to maintain.
Can we just inline it locally in `HttpChannelState`, which is an internal class?